src/StructuredDiscussions/SDTopic.php

```diff
@@ -81,8 +81,15 @@ public function getSummary(): ?string {
 	 *
 	 * @return string
 	 */
-	public function getCreator(): string {
-		return $this->getRootRevision()['creator']['name'];
+	public function getCreator(): ?string {
+		$creator = $this->getRootRevision()['creator']['name'];
```

Comment:
Maybe something like
```php
$rootRevision = $this->getRootRevision();
if (
    !isset( $rootRevision[ 'creator' ] )
    || !isset( $rootRevision[ 'creator' ][ 'name' ] )
    || !is_string( $rootRevision[ 'creator' ][ 'name' ] )
) {
    return null;
}

return $rootRevision[ 'creator' ][ 'name' ];
```
for less notices?

Reply:
The error from #11 creates the impression that the value is still set (otherwise, it would have been `NULL`).

P.S. PHP also supports this: `return $rootRevision['creator']['name'] ?? null`.

Reply:
> The error from https://github.com/WikibaseSolutions/SemanticStructuredDiscussions/issues/11 creates the impression that the value is still set

Ah yes indeed
> P.S. PHP also supports this: `return $rootRevision['creator']['name'] ?? null`

So do you want to do `return ($rootRevision['creator']['name'] ?? null) ?: null` then? Sounds good to me